### PR TITLE
lc-compile: Add -Werror option and use it everywhere.

### PR DESCRIPTION
**Pull request by peter-b**
Add a `-Werror` option to the LCB compiler, which turns all warnings into errors.  Use it whenever compiling LCB source code that's part of LiveCode.

### CODE REVIEW

File: extensions/widgets/browser/browser.lcb [hunk 1]
```diff
@@ -24,14 +24,14 @@ Licensed under the terms of the Creative Commons Attribution 3.0 Unported Licens
...
 This widget displays web content within a native web browser view.
 
-Name: url
+Name: URL
 Type: property
 
 Syntax:
-set the url of <widget> to <pUrl>
-get the url of <widget>
+set the URL of <widget> to <pUrl>
```
Comment (runrevmark) [inline extensions/widgets/browser/browser.lcb:31]: Why are these uppercased? Remember that properties are only accessible from LCS, and there property names are camel-cased.

Comment (runrevmark) [inline extensions/widgets/browser/browser.lcb:31]: Okay - so 'url' is somewhat inconsistent in LCS at the moment... I vote for normalizing around the standard convention which is camel-case - so 'url', not 'Url' or 'URL'. (Things are case-insensitive anyway, however we need to make sure we keep to a convention).

Comment (peter-b) [inline extensions/widgets/browser/browser.lcb:31]: They're uppercased because [bug 14980](http://quality.livecode.com/show_bug.cgi?id=14980)

Comment (runrevmark) [inline extensions/widgets/browser/browser.lcb:31]: I retract my previous concern - URL is inconsistently cased in the current dictionary and so this is as good a choice as any.

**Comment by runrevmark** [comment]
@livecode-vulcan review ok ed3ec7e

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok ed3ec7e5af5ffb6be6a40e733287d489350ddada

**Comment by livecode-vulcan** [comment]
😎 test success ed3ec7e5af5ffb6be6a40e733287d489350ddada
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1063)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1072)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1060)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1065)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1076)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1081)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1081)

**Pull request merged**